File: src/dataset/javascript/rna-transcription_exercism-javascript-oguzsh/gemini/GeminiFlash_rna-transcription_v2.js
export const toRna = (dna = "") => {
  const dna_rna_complement = {
    G: "C",
    C: "G",
    T: "A",
    A: "U",
  };

  let rna_strands = "";
  for (let i = 0; i < dna.length; i++) {
    rna_strands += dna_rna_complement[dna[i]];
  }
  return rna_strands;
};

File: src/dataset/c/rna-transcription_exercism-leleah/gemini/GeminiFlash_rna_transcription_v2.c
#include "rna_transcription.h"
#include <stdlib.h>
#include <string.h>

char *to_rna(const char *dna) {
  size_t dna_len = strlen(dna);
  char *rna = malloc(dna_len + 1);
  if (!rna) return NULL; 

  for (size_t i = 0; i < dna_len; ++i) {
    switch (dna[i]) {
      case 'G':
        rna[i] = 'C';
        break;
      case 'C':
        rna[i] = 'G';
        break;
      case 'T':
        rna[i] = 'A';
        break;
      case 'A':
        rna[i] = 'U';
        break;
      default:
          free(rna);
          return NULL;
    }
  }
  rna[dna_len] = '\0';
  return rna;
}